hw-cliargs/ls_helper.py

```diff
@@ -0,0 +1,84 @@
+"""Helper classes for ls utility"""
+
+from collections import OrderedDict
+import functools
+
+
+class EntryInfo:
+    """Class to store entries information and provide printable representation for it"""
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+    def __repr__(self):
+        return '  '.join(f'{k}: {v}' for k, v in self.__dict__.items())
+
+    def build_printable(self,
+                        pretty_size: bool = False,
+                        show_time: str = 'ctime',
+                        long_format: bool = False):
+        """Depending on required format returns EntryInfo representation
+
+        Args:
+          pretty_size: set up human-readable format for file size if True,
+            otherwise set up raw format in bytes
+
+          show_time: set up what time to show. One of (`atime`, `ctime`, `mtime`)
+
+          long_format: set up full format (time, mode, size, name) if True,
+            otherwise set up short format (name)
+        """
+
+        if show_time not in ('atime', 'ctime', 'mtime'):
+            raise ValueError('Only (`atime`, `ctime`, `mtime`) allowed for show_time argument')
+
+        if long_format:
+            return '  '.join(
+                [
+                    self.__dict__.get(show_time),
+                    self.__dict__.get('mode'),
+                    f"{str(self.__dict__.get('size_pretty' if pretty_size else 'size'))}",
+                    self.__dict__.get('name')
+                ]
+            )
+        else:
+            return self.__dict__.get('name')
```

Comment:
I would check it at the beginning, so you should not indent all the code inside if